pinot-server/src/main/java/org/apache/pinot/server/starter/ServerInstance.java

```diff
@@ -38,36 +45,64 @@
 public class ServerInstance {
   private static final Logger LOGGER = LoggerFactory.getLogger(ServerInstance.class);
 
-  private ServerMetrics _serverMetrics;
-  private InstanceDataManager _instanceDataManager;
-  private QueryExecutor _queryExecutor;
-  private QueryScheduler _queryScheduler;
-  private QueryServer _queryServer;
-  private LongAccumulator _latestQueryTime;
+  private final ServerMetrics _serverMetrics;
+  private final InstanceDataManager _instanceDataManager;
+  private final QueryExecutor _queryExecutor;
+  private final LongAccumulator _latestQueryTime;
+  private final QueryScheduler _queryScheduler;
+  private final QueryServer _queryServer;
 
   private boolean _started = false;
 
-  public void init(ServerConf serverConf, ZkHelixPropertyStore<ZNRecord> propertyStore)
+  public ServerInstance(ServerConf serverConf, HelixManager helixManager)
       throws Exception {
     LOGGER.info("Initializing server instance");
 
-    ServerBuilder serverBuilder = new ServerBuilder(serverConf, propertyStore);
-    _serverMetrics = serverBuilder.getServerMetrics();
-    _instanceDataManager = serverBuilder.buildInstanceDataManager();
-    _queryExecutor = serverBuilder.buildQueryExecutor(_instanceDataManager);
+    LOGGER.info("Initializing server metrics");
```

Comment:
Nit: int()?

Reply:
I prefer constructor over init() as we can make member variables final and immutable